GameServerLib/GameObjects/AttackableUnits/AI/LaneMinion.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using GameServerCore.Domain.GameObjects;
+using GameServerCore.Enums;
+using LeagueSandbox.GameServer.GameObjects.Stats;
+
+namespace LeagueSandbox.GameServer.GameObjects.AttackableUnits.AI
+{
+    public class LaneMinion : Minion, ILaneMinion
+    {
+        /// <summary>
+        /// Const waypoints that define the minion's route
+        /// </summary>
+        protected List<Vector2> _mainWaypoints;
+        protected int _curMainWaypoint;
+        public MinionSpawnPosition SpawnPosition { get; }
+        public MinionSpawnType MinionSpawnType { get; }
+        protected bool _aiPaused;
+
+        public LaneMinion(
+            Game game,
+            MinionSpawnType spawnType,
+            MinionSpawnPosition position,
+            List<Vector2> mainWaypoints,
+            uint netId = 0
+        ) : base(game, null, 0, 0, "", "", 1100, netId)
```

Comment:
Do you really want to pass null, why?

Reply:
I didn't know any alternatives, perhaps it should be Nexus or barracks, is that right?

Reply:
`null` is bad usually, what you want is to pass some default object, it depends on the context.

Reply:
`null` here stands for the owner. The owner should be the object itself probably or the barracks created it, depends on packets. Currently I think null is ok, but we should change Minion class to accept no owner.